rust/connlib/tunnel/src/sockets.rs

```diff
@@ -292,44 +293,55 @@ impl Socket {
     }
 
     fn poll_flush(&mut self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
-        loop {
-            match self.socket.try_io(Interest::WRITABLE, || {
-                self.state
-                    .send((&self.socket).into(), &self.buffered_transmits)
-            }) {
-                Ok(0) => break,
-                Err(e) if e.kind() == io::ErrorKind::WouldBlock => break,
-                Err(e) => return Poll::Ready(Err(e)),
-
-                Ok(num_sent) => {
-                    self.buffered_transmits.drain(..num_sent);
+        while let Some(transmit) = self.buffered_transmits.front() {
```

Comment:
```suggestion
        // Only pop if we successfully send it
        while let Some(transmit) = self.buffered_transmits.front() {
```
I expected it to be `pop_front` but I see that's inside the loop